packages/kit/src/runtime/app/navigation.js

```diff
@@ -16,6 +16,7 @@ export const goto = import.meta.env.SSR ? guard('goto') : goto_;
 export const invalidate = import.meta.env.SSR ? guard('invalidate') : invalidate_;
 export const prefetch = import.meta.env.SSR ? guard('prefetch') : prefetch_;
 export const prefetchRoutes = import.meta.env.SSR ? guard('prefetchRoutes') : prefetchRoutes_;
+export const onBeforeNavigate = import.meta.env.SSR ? guard('onBeforeNavigate') : onBeforeNavigate_;
```

Comment:
If you change this line to be 
```javascript
export const onBeforeNavigate = import.meta.env.SSR ? () => {} : onBeforeNavigate_;
```
you no longer need to wrap the whole thing in ``if (browser)`` since the guard won't throw an error when SSRing.

I understand why you would like this API without a browser check, as that would make it behave similar to onMount. On the other hand, this would cause onBeforeNavigate to fail silently when users tried to run it in an SSR only context. Adding a console.warn would not work for this either, since even if used correctly, it would warn the user.

I guess if we were to go the no ``if (browser)`` route, solid documentation for this API would be needed to communicate this.

Reply:
Yes, I think it will be easier if we remove the if (browser) check. I've added a commit that removes it, thanks!